infra/src/main/resources/db/migration/repeatable/R__04.06_track_number_in_layout_context.sql

```diff
@@ -41,13 +39,16 @@ select
                             from layout.track_number overriding_draft
                             where overriding_draft.design_id is not distinct from design_id_in
                               and overriding_draft.draft
-                              and (overriding_draft.official_row_id = row.id or
-                                   overriding_draft.design_row_id = row.id))
+                              and row.id = case
+                                             when design_id_in is null then overriding_draft.official_row_id
+                                             else overriding_draft.design_row_id
```

Comment:
Case: ollaan tekemässä uutta designia ja pyydetään draft-design riviä sellaiselle käsittelle joka on vain officialissa. 
Eikös tässä tuolloin pitäisi katsoa että `row.id = coalesce(overriding_draft.design_row_id, overriding_draft.official_row_id)`.

Reply:
Tämä Jyrkin tapa kuulostaisi munkin korvaani oikeammalta. Tai vähintäänkin jossain se pitää tehdä näin ennen lopullisten tulosten hakua, ja tämä vaikuttaisi järkevältä paikalta